src/logger.rs

```diff
@@ -0,0 +1,58 @@
+use std::{env, sync::OnceLock};
+
+#[derive(Debug, Clone, Copy)]
+enum FfmpegLogLevel {
+    Error,
+    Warn,
+    Info,
+    Debug,
+}
+
+fn ffmpeg_log_level() -> FfmpegLogLevel {
+    static LOG_LEVEL: OnceLock<FfmpegLogLevel> = OnceLock::new();
+
+    *LOG_LEVEL.get_or_init(|| match env::var("FFMPEG_LOG_LEVEL") {
+        Ok(debug) if debug == "debug" => FfmpegLogLevel::Debug,
+        Ok(info) if info == "info" => FfmpegLogLevel::Info,
+        Ok(warn) if warn == "warn" => FfmpegLogLevel::Warn,
+        Ok(error) if error == "error" => FfmpegLogLevel::Error,
+        Ok(_) => FfmpegLogLevel::Warn,
+        Err(_) => FfmpegLogLevel::Warn,
+    })
+}
+
+extern "C" fn ffmpeg_log_callback(
+    arg1: *mut libc::c_void,
+    log_level: libc::c_int,
+    fmt: *const libc::c_char,
+    va_list_tag: *mut ffmpeg_next::sys::__va_list_tag,
+) {
+    unsafe {
+        match ffmpeg_log_level() {
+            FfmpegLogLevel::Error if log_level <= 16 => {
+                ffmpeg_next::sys::av_log_default_callback(arg1, log_level, fmt, va_list_tag)
+            }
+            FfmpegLogLevel::Warn if log_level <= 24 => {
+                ffmpeg_next::sys::av_log_default_callback(arg1, log_level, fmt, va_list_tag)
+            }
+            FfmpegLogLevel::Info if log_level <= 32 => {
+                ffmpeg_next::sys::av_log_default_callback(arg1, log_level, fmt, va_list_tag)
+            }
+            FfmpegLogLevel::Debug if log_level <= 48 => {
+                ffmpeg_next::sys::av_log_default_callback(arg1, log_level, fmt, va_list_tag)
+            }
+            _ => (),
+        }
+    }
+}
+
+pub fn init_logger() {
+    if env::var("CARGO_MANIFEST_DIR").is_ok() {
```

Comment:
Shouldn't we also have a manual switch for this?

Reply:
I'll be reworking env based configuration in follow-up PR, so I can address it then.